src/Microsoft.Azure.WebJobs.Host/Config/FluentBindingRule.cs

```diff
@@ -0,0 +1,311 @@
+
+            return BindToCollector(converter);
+        }
+
+        /// <summary>
+        /// Bind to a collector. 
+        /// </summary>
+        /// <typeparam name="TMessage"></typeparam>
+        /// <param name="buildFromAttribute"></param>
+        /// <returns></returns>
+        public FluentBindingRule<TAttribute> BindToCollector<TMessage>(
+           IConverter<TAttribute, IAsyncCollector<TMessage>> buildFromAttribute)
+        {
+            var bf = _parent.BindingFactory;
+            var pm = PatternMatcher.New(buildFromAttribute);
+            var rule = new AsyncCollectorBindingProvider<TAttribute, TMessage>(bf.NameResolver, bf.ConverterManager, pm);
+
+            return Bind(rule);
+        }
+
+        /// <summary>
+        /// Bind to a collector. 
+        /// </summary>
+        /// <typeparam name="TMessage"></typeparam>
+        /// <param name="builderType"></param>
+        /// <param name="constructorArgs"></param>
+        /// <returns></returns>
+        public FluentBindingRule<TAttribute> BindToCollector<TMessage>(
+             Type builderType,
+             params object[] constructorArgs)
+        {
+            var bf = _parent.BindingFactory;
+            var pm = PatternMatcher.New(builderType, constructorArgs);
+            var rule = new AsyncCollectorBindingProvider<TAttribute, TMessage>(bf.NameResolver, bf.ConverterManager, pm);
+            return Bind(rule);
+        }
+
+        #endregion // BindToCollector
+
+        /// <summary>
+        /// Setup a trigger binding for this attribute
+        /// </summary>
+        /// <param name="trigger"></param>
+        public void BindToTrigger(ITriggerBindingProvider trigger)
+        {
+            if (_binders.Count > 0)
+            {
+                throw new InvalidOperationException($"The same attribute can't be bound to trigger and non-trigger bindings");
+            }
+            IExtensionRegistry extensions = _parent.GetService<IExtensionRegistry>();
+            extensions.RegisterExtension<ITriggerBindingProvider>(trigger);
+        }
+
+        /// <summary>
+        /// Add a validator for the set of rules. 
+        /// The validator will apply to all of these rules. 
+        /// </summary>
+        /// <param name="validator"></param>
+        /// <returns></returns>
+        [Obsolete("move this directly onto the attribute")]
+        public FluentBindingRule<TAttribute> AddValidator(Action<TAttribute, Type> validator)
+        {
+            if (_validator != null)
+            {
+                throw new InvalidOperationException("Validator already set");
+            }
+            _validator = validator;
+            return this;
+        }
+
+        // Called by infrastructure after the extension is invoked.
+        // This applies all changes accumulated on the fluent object. 
+        internal void Flush()
+        {
```

Comment:
Flush seems like the wrong name -- could this be something like Build or Initialize?